src/diff/index.js

```diff
@@ -406,7 +406,8 @@ export function unmount(vnode, parentVNode, skipRemove) {
 	if (options.unmount) options.unmount(vnode);
 
 	if ((r = vnode.ref)) {
-		applyRef(r, null, parentVNode);
+		if (!r.current || r.current === vnode._dom)
```

Comment:
Would this work? Seems like we're wanting to null the ref if it's not already null.

```suggestion
		if (r.current != null)
```

Reply:
We only want to nul lthe ref if it's assigned to the node that's unmounting else we could result in a false positive (see issue #2194)